hnix-store-core/src/System/Nix/Internal/Hash.hs

```diff
@@ -76,6 +76,7 @@ class HasDigest (a :: HashAlgorithm) where
   initialize        :: AlgoCtx a
   update            :: AlgoCtx a -> BS.ByteString -> AlgoCtx a
   finalize          :: AlgoCtx a -> Digest a
+  hashName          :: T.Text
```

Comment:
This is a sensible place to put `hashName` (I think others have put this under its own typeclass, but simply using `HashDigest` seems better). `HasDigest` is not a great typeclass name under this change though. What's a name for this class that's neither (a) too specific for the methods it provides, (b) as general as `"Hash"`?

Reply:
I disagree that this should be under `HasDigest` (which I've renamed `ValidAlgo` in my branch). Not every hash has a canonical name, e.g. the truncated ones don't (and shouldn't!).